Compiler.Lexer/Compiler.Lexer/Tokenizer.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Compiler.Lexer
+{
+    public partial class Tokenizer
+    {
+        private CharStream Stream;
+        private Token LastToken;
+
+        public Tokenizer(string source)
```

Comment:
There should be a default Tokenizer() constructor with no parameters. Otherwise .NET will automatically make one for you. Otherwise the class can be instantiated without a value for source and error when methods referencing it are called. Make the default constructor private so the existing constructor with the "source" variable passed in has to be used.